### PR TITLE
Make sure frequently *read* cache entries never expire

### PR DESCRIPTION
**Pull request by stkem**
@ymatsuda @raymondng have a look

### CODE REVIEW

File: kifi-backend/modules/common/app/com/keepit/common/cache/FortyTwoCache.scala [hunk 1]
```diff
@@ -240,6 +248,12 @@ trait FortyTwoCache[K <: Key[T], T] extends ObjectCache[K, T] {
           errorMessage = Some(s"Failed fetching key $key from $repo")))
         None
     }
+    if (valueOpt.isDefined && Random.nextFloat>0.99) {
```
Comment (yingjieMiao) [inline kifi-backend/modules/common/app/com/keepit/common/cache/FortyTwoCache.scala]: interesting approach. Note 0.99^100 ~ 0.366. If I understand the code correctly, if the cache is hit 100 times, there is still 36.6% chance it is not touched at all.

Comment (stkem) [inline kifi-backend/modules/common/app/com/keepit/common/cache/FortyTwoCache.scala]: Yup. Basically the idea is that entries that get accessed very frequently within their initially set expiration lifetime will get touched with high probability thus resetting the expiration time (which `get` by itself does not do). The number 0.99 is pretty arbitrary and we may have to adjust it. The goal was not to get a touch every 100 `get`s.

Comment (gmethvin) [inline kifi-backend/modules/common/app/com/keepit/common/cache/FortyTwoCache.scala]: Cool. Can we factor out this magic number into a constant with a meaningful name?

File: kifi-backend/modules/common/app/com/keepit/common/cache/FortyTwoCache.scala [hunk 2]
```diff
@@ -240,6 +248,12 @@ trait FortyTwoCache[K <: Key[T], T] extends ObjectCache[K, T] {
...
+      ttl match {
```
Comment (gmethvin) [inline kifi-backend/modules/common/app/com/keepit/common/cache/FortyTwoCache.scala]: can't we just say

``` scala
if (ttl.isFinite) repo.touch(key.toString, ttl.toSeconds.toInt)
```

?

Comment (gmethvin) [inline kifi-backend/modules/common/app/com/keepit/common/cache/FortyTwoCache.scala]: Also, while you're editing this file, you could also change the similar expression in `FortyTwoCache#setInnerCache` to

``` scala
val ttlInSeconds = if (ttl.isFinite) ttl.toSeconds.toInt else 0
```

Comment (stkem) [inline kifi-backend/modules/common/app/com/keepit/common/cache/FortyTwoCache.scala]: Good point. Was not aware of the `isFinite` method.

**Comment by andrewconner** [comment]
@stkem Is this wanting to get in still? Can't be auto-merged.

**Comment by stkem** [comment]
@andrewconner  I'm not so sure any more that this is a good idea. Especially since we have support for key with no expiration now. I'm going to close it for now. If we see the need arising we can get it back quickly.

**Pull request closed without merging**